### PR TITLE
Added parameter for app display name for dynamic rendering in the wallet during web5 connect flow

### PR DESCRIPTION
**Pull request by thehenrytsai**
Currently there is no way for the wallet to dynamically display to the end-user a friendly display name when displaying the permission request. This PR introduces an app specified property for this purpose.

Out of scope for this PR, but a few items to consider for the future.
1. Consider adding a friendly name property in protocol definition for even better UX.
1. UX  would be simpler/cleaner if app just asks delegated permission for a protocol, not enumerating through all the read, query, write etc.
1. If the permission request (coming from a website) is signed by a `did:web` DID or similar instead of an ephemeral DID, we'd be able to go the extra mile of perhaps matching the display name with the domain name and give user more assurances.

Tested against modified react starter and my own wallet.

PS. Kept some logging I added while debugging, happy to remove if preferred.

### CODE REVIEW

**Review by csuwildcat — APPROVED** [review]

**Review by LiranCohen — CHANGES_REQUESTED** [review]
Great work, but agree with Tim I don't think we need to include a ton of logging here.

However I do think we should include some optional callbacks in various state of the flow in order to allow a wallet/app to display status changes as things are happening and/or log something themselves when testing the flow.

**Comment by thehenrytsai** [comment]
Good call on keeping console log clean by default. Was hoping to wait until we have a more thought-through logger to use, but added a logger that doesn't log by default, we can iterate over the logger as needed.

Happy to separate the logger portion into a separate PR, kept the logs in this PR to show the context why/when they were introduced.

Optional callbacks in various state of the flow would be cool, but probably scope for a different PR.

File: packages/agent/src/oidc.ts [hunk 1]
```diff
@@ -790,6 +804,7 @@ async function submitAuthResponse(
   });
 
   // Sign the Response Object using the ephemeral DID's signing key.
+  console.log('Signing auth response object...');
```
[the remark this answers is not in the document]
Comment (thehenrytsai) [reply packages/agent/src/oidc.ts]: You are right it is not directly related to this PR. Was added when I was testing/debugging the change end to end.

**Comment by thehenrytsai** [comment]
@shamilovtim, Imma rip this out into a separate PR since the main goal is go rid of hardcoded app name in the wallet.

I will say that I tested the logger works in browser, but taking in your suggestion, I'll just hand-role a few lines for the implementation without depending on an external lib, unless you have one that you want to recommend. The logger interface is there for exactly the purpose of swapping out implementation easily.

File: packages/agent/src/oidc.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (thehenrytsai) [reply packages/agent/src/oidc.ts]: Log removed.

**Review by LiranCohen — APPROVED** [review]

**Pull request merged**